pyccel/ast/headers.py

```diff
@@ -510,7 +497,7 @@ def __reduce_ex__(self, i):
            and its arguments
            """
 
-        args = (self.name,
+        args = (self.name.split('.'),
```

Comment:
Was there a bug here?

Reply:
I hadn't realised it but yes. `__reduce_ex__` should return a tuple which contains valid arguments for the class. `MethodHeader.__init__` requires the argument `name` to be a tuple or a list not a string

Reply:
Well done! Should we add a note about it in the PR's description?